brayns/io/NESTLoader.cpp

```diff
@@ -0,0 +1,253 @@
+    {
+        BRAYNS_ERROR << "Failed to create cache file" << std::endl;
+        return false;
+    }
+
+    // Write header
+    simulationHandler->setNbFrames( nbFrames );
+    simulationHandler->setFrameSize( _frameSize );
+    simulationHandler->writeHeader( file );
+
+    BRAYNS_INFO << "Spike report contains " << nbFrames << " frames of "
+                << _frameSize << " values each" << std::endl;
+
+    // Write body
+    for( uint64_t frame = 0; frame < nbFrames; ++frame )
+    {
+        _load( frame );
+        simulationHandler->writeFrame( file, _spikingTimes );
+    }
+    file.close();
+
+    scene.setSimulationHandler( simulationHandler );
+
+    BRAYNS_INFO << "----------------------------------------" << std::endl;
+    BRAYNS_INFO << "Number of frames: " << nbFrames << std::endl;
+    BRAYNS_INFO << "Frame size      : " << _frameSize << std::endl;
+    BRAYNS_INFO << "----------------------------------------" << std::endl;
+    return true;
+}
+
+bool NESTLoader::_loadBinarySpikes( const std::string& spikesFilename )
+{
+    std::ifstream file( spikesFilename, std::ios::out | std::ios::binary );
+    file.seekg( 0, std::ios::end );
+    const size_t fileSize = file.tellg();
+    file.seekg( 0 );
+
+    _nbElements = ( fileSize - NEST_HEADER_SIZE ) / ( sizeof( uint32_t ) + sizeof( float ));
+    BRAYNS_INFO << "Loading " << _nbElements << " spikes from " << spikesFilename << std::endl;
+
+    // Parse header
+    uint32_t magic;
+    file.read(( char* )&magic, sizeof( uint32_t ));
+
+    if( NEST_MAGIC != magic )
+        return false;
+
+    uint32_t version;
+    file.read(( char* )&version, sizeof( uint32_t ));
+    if( NEST_VERSION != version )
+        return false;
+
+    // Parse elements
+    uint32_t gid;
+    float value;
+    _values.reserve( _nbElements );
+    _gids.reserve( _nbElements );
+    size_t i = 0;
+    while( !file.eof( ))
+    {
+        BRAYNS_PROGRESS( i, _nbElements );
+        file.read(( char* )&value, sizeof( float ));
+        _values.push_back( value );
+        file.read(( char* )&gid, sizeof( uint32_t ));
+        _gids.push_back( gid );
+        ++i;
+    }
+
+    _spikesStart = _values[ 0 ]; // first spike timestamp after header
+    _spikesEnd = _values[ _nbElements - 1 ]; // last spike timestamp
+
+    BRAYNS_INFO << "Loaded " << i << " elements out of " << _nbElements << std::endl;
+    BRAYNS_INFO << "Spikes interval: [" << _spikesStart << " - " << _spikesEnd << "]" << std::endl;
+
+    file.close();
+    return true;
+}
+
+bool NESTLoader::_load( uint64_t frame )
+{
+    const float start = frame * NEST_TIMESTEP;
+    const float end = ( 1 + frame ) * NEST_TIMESTEP;
+
+    BRAYNS_DEBUG << "Loading spikes at frame: " << frame
+                 << " (" << start << " ms)" << std::endl;
+
+    size_t nbSpikes = 0;
+    for( size_t i = 0; i < _nbElements; ++i )
+    {
+        const float time = _values[ i ];
+        // whith the next check, simulation only plays forward
```

Comment:
I did it like this so the loading could be faster, considering only timestamps bigger than current. But now you can only play the simulation forward. If that's ok it should be documented.